asv_bench/benchmarks/tslibs/timedelta.py

```diff
@@ -0,0 +1,61 @@
+"""
+Timedelta benchmarks that rely only on tslibs.  See benchmarks.timedeltas for
+Timedelta benchmarks that rely on other parts fo pandas.
+"""
+import datetime
+
+import numpy as np
+
+from pandas import Timedelta
+
+
+class TimedeltaConstructor:
+    def time_from_int(self):
+        Timedelta(123456789)
+
+    def time_from_unit(self):
+        Timedelta(1, unit="d")
+
+    def time_from_components(self):
+        Timedelta(
+            days=1,
+            hours=2,
+            minutes=3,
+            seconds=4,
+            milliseconds=5,
+            microseconds=6,
+            nanoseconds=7,
+        )
+
+    def time_from_datetime_timedelta(self):
+        Timedelta(datetime.timedelta(days=1, seconds=1))
+
+    def time_from_np_timedelta(self):
+        Timedelta(np.timedelta64(1, "ms"))
+
+    def time_from_string(self):
+        Timedelta("1 days")
+
+    def time_from_iso_format(self):
+        Timedelta("P4DT12H30M5S")
+
+    def time_from_missing(self):
+        Timedelta("nat")
+
+
+class TimedeltaProperties:
+    def setup_cache(self):
```

Comment:
Orthogonal but maybe easy to change here if you want - using `setup_cache` here seems overkill. Mostly used for longer running setup calls, which I can't imagine this is

Reply:
generally agreed, but for now prefer to keep this move-only.

Longer-term I hope to get asv to use something like setup_class so that it only gets run once instead of many times